98 design doc plot bounds (#144)
Genentech/phase1b: design/design_doc_plotBounds.qmd

---
title: "Design Doc for plotBounds"
format: revealjs
editor: visual
---

> Statisticians can display boundaries of Go, Stop and GreyZone decisions graphically in terms of response rates required by desired samples sizes or looks. This is such purpose of `plotBounds`. Note that for the predictive probability case, decision 1 rules are chosen. See `ocPredprob` for rules. Design document aims to translate previous call function to use of more common packages such as `ggplot2` and reduce number of lines. 

Reviewer: ```suggestion
Statisticians can display boundaries of Go, Stop and GreyZone decisions graphically in terms of response rates required by desired samples sizes or looks. This is the purpose of the  `plotBounds` function. Coordinates are the points where the polygon should be drawn and reflect the possible values for a go and stop decision respectively.

 Note that for the predictive probability case, decision 1 rules are chosen. See `ocPredprob` for rules. This design document aims to refactor the previous function, in particular to use `ggplot2` and reduce the length of the function. 
```

    ymax_stop = coords$xL
  )
  plot <- ggplot2::ggplot() +
    ggplot2::geom_ribbon(
      data = stop_data,
      ggplot2::aes(x = n, ymin = ymin_stop, ymax = ymax_stop),
      fill = stop_colour, alpha = alpha
    ) +
    ggplot2::geom_ribbon(
      data = go_data,
      ggplot2::aes(x = n, ymin = ymin_go, ymax = ymax_go),
      fill = go_colour, alpha = alpha
    ) +
    ggplot2::scale_x_continuous(
      name = "n",
      breaks = go_data$n
    ) +
    ggplot2::scale_y_continuous(
      breaks = seq(from = 0, to = max(go_data$n), by = 2)
    ) +
    ggplot2::geom_point(
      data = go_data,
      ggplot2::aes(x = n, y = ymin_go),
      size = 3,
      shape = 16,
      colour = "#16C47F",
    ) +
    ggplot2::geom_point(
      data = stop_data,
      ggplot2::aes(x = n, y = ymax_stop),
      size = 3,
      shape = 16,
      colour = "#FF748B", # "#DC3C22",FF748B
    ) +
    ggplot2::xlab("Sample Size (n)") +
    ggplot2::ylab("Number of Responders") +
    ggplot2::theme_minimal()

  plot + ggplot2::ggtitle(
    "Boundaries of Go and Stop Responders per sample size"
  )
}
```

# creating plot using new function with output from `boundsPredprob()`

> The following output show that important coordinates needed as input to create these plots. 

Reviewer: ```suggestion
The following example shows which inputs from a predictive probability design are needed to create these plots. 
```

As a premise, we establish the criteria:

Interim
Go : P(success) > 90 %
Stop : P(success) > 10 %

Final
Go : P(RR > 0.20 | data) > 80 %
Stop : P(RR > 0.20 | data) < 10 %

The criteria for interim analyses are from predictive probability calculation.

The criteria for final analyses are from posterior probability calculation. 

which translates to boundaries :
When n = 10, 4 responders and above is a Go, 0 responders and below is a stop.
When n = 20, 7 responders and above is a Go, 2 responders and below is a stop
When n = 30, 9 responders and above is a Go, 5 responders and below is a stop

> When n = 40, 10 responders and above is a Go, 9 responders and below is a stop

Reviewer: ```suggestion
which translates to boundaries :

- When n = 10, 4 responders and above is a Go, 0 responders and below is a stop.
- When n = 20, 7 responders and above is a Go, 2 responders and below is a stop
- When n = 30, 9 responders and above is a Go, 5 responders and below is a stop
- When n = 40, 10 responders and above is a Go, 9 responders and below is a stop
```

```{r}
coords <- boundsPredprob(
  looks = c(10, 20, 30, 40),
  p0 = 0.20,
  tT = 0.80,
  phiL = 0.10,
  phiU = 0.90,
)
print(coords)
```

```{r}
coords <- boundsPredprob(
  looks = c(10, 20, 30, 40),
  p0 = 0.20,
  tT = 0.80,
  phiL = 0.10,
  phiU = 0.90,
)
new_plotBounds(
  coords = coords,
  go_colour = "blue",
  alpha = 1
)
```

# creating plots with new function with output from `boundsPostprob()`

> The following output show that important coordinates needed as input to create these plots. Coordinates are the points where the polygon should be drawn and reflect the possible values for a go and stop decision respectively.

Reviewer: ```suggestion
The following output show which inputs from a posterior probability design are needed to create these plots.
```

As a premise, we establish the criteria:
Go : P(RR > 0.20 | data) > 90 %
Stop : P(RR > 0.20 | data) > 10 %
which translates to boundaries :
When n = 10, 4 responders and above is a Go, 3 responders and below is a stop.
When n = 20, 7 responders and above is a Go, 6 responders and below is a stop
When n = 30, 9 responders and above is a Go, 8 responders and below is a stop

> When n = 40, 12 responders and above is a Go, 11 responders and below is a stop

Reviewer: ```suggestion
which translates to boundaries :

- When n = 10, 4 responders and above is a Go, 3 responders and below is a stop.
- When n = 20, 7 responders and above is a Go, 6 responders and below is a stop
- When n = 30, 9 responders and above is a Go, 8 responders and below is a stop
- When n = 40, 12 responders and above is a Go, 11 responders and below is a stop
```

```{r}
coords1 <- boundsPostprob(
  looks = c(10, 20, 30, 40),
  p0 = 0.20,
  tL = 0.10,
  tU = 0.90,
  parE = c(1, 1)
)
new_plotBounds(coords1)
```

> 

Reviewer: please discuss here whether it could be useful to refactor this in terms of a `plot()` method for S3 classes `boundsPostprob` and `boundsPredprob`

Author: thanks @danielinteractive, let me know if I understood correctly

Reviewer: did you discuss this already?

Author: yes, on line 157, sorry - can't permalink it to here somehow

Reviewer: ah I see, yeah but this is not what I had in mind - my idea was to have `plot` methods for objects returned by the bounds functions. so I was curious to have some thoughts in the design doc about pros/cons of that vs. usual functions to do the plots

Reviewer: let's keep it as is for now, I will open a separate issue to think about this more generally for the package
